src/Neo/Wallets/Wallet.cs

```diff
@@ -657,6 +660,36 @@ public bool Sign(ContractParametersContext context)
             return fSuccess;
         }
 
+        /// <summary>
+        /// Signs the specified data with the corresponding private key of the specified public key.
+        /// </summary>
+        /// <param name="signData">The data to sign.</param>
+        /// <param name="publicKey">The public key.</param>
+        /// <returns>The signature.</returns>
+        public byte[] SignWithOwnedKeyPair(byte[] signData, ECPoint publicKey)
```

Comment:
```suggestion
        public byte[] Sign(byte[] signData, ECPoint publicKey)
```

It's obvious that it is `KeyPair` seeing how it's in the `KeyPair` class.

Reply:
You have an object of `KeyPair` which has a private key, all within a wallet. When you are to `sign` something you have to use private key, `context.ScriptHashes`  is used to lookup the account to get its private key. I expect this method to do the same as `Sign` does. Since it does with the only difference being you are using a public key to lookup the account to get the private key. You can easily create a `scripthash` with a public key. So what's the problem with not having the same name? It does just that, **SIGNS** data, naming it `Sign` is appropriate and would be expected by anyone not knowing the API. What is `OwnedKeyPair`? One would ask themself, Its not returning `KeyPair` nor is it wanting a `KayPair`. It is in the `KeyPair` class, so what does it mean?